cub/cmake/CubCudaConfig.cmake

```diff
@@ -4,19 +4,16 @@ enable_language(CUDA)
 # Architecture options:
 #
 
-# Since we have to filter the arch list based on target features, we don't
-# currently support the convenience arch flags:
-if ("all" IN_LIST CMAKE_CUDA_ARCHITECTURES OR
-    "all-major" IN_LIST CMAKE_CUDA_ARCHITECTURES OR
-    "native" IN_LIST CMAKE_CUDA_ARCHITECTURES)
-  message(FATAL_ERROR
-    "The CUB dev build requires an explicit list of architectures in CMAKE_CUDA_ARCHITECTURES. "
-    "The convenience flags of 'all', 'all-major', and 'native' are not supported.\n"
-    "CMAKE_CUDA_ARCHITECTURES=${CMAKE_CUDA_ARCHITECTURES}")
-endif()
-
 # Create a new arch list that only contains arches that support CDP:
-set(CUB_CUDA_ARCHITECTURES ${CMAKE_CUDA_ARCHITECTURES})
+if ("native" IN_LIST CMAKE_CUDA_ARCHITECTURES)
+  set(CUB_CUDA_ARCHITECTURES ${CMAKE_CUDA_ARCHITECTURES_NATIVE})
```

Comment:
Can you point me towards the docs for this? I can't find any information about `CMAKE_CUDA_ARCHITECTURES_NATIVE`.

Reply:
I found no docs for this, just this forum entry: https://gitlab.kitware.com/cmake/cmake/-/issues/24921

Reply:
@robertmaynard Can we rely on these? It'd greatly simplify our lives if we can!

Reply:
They are internal variables and subject to change without notice.